src/main/java/com/limechain/chain/lightsyncstate/scale/AuthorityReader.java

```diff
@@ -4,11 +4,10 @@
 import io.emeraldpay.polkaj.scale.ScaleCodecReader;
 import io.emeraldpay.polkaj.scale.ScaleReader;
 import io.emeraldpay.polkaj.scale.reader.UInt64Reader;
-import io.emeraldpay.polkaj.types.Hash256;
 
 public class AuthorityReader implements ScaleReader<Authority> {
     @Override
     public Authority read(ScaleCodecReader reader) {
-        return new Authority(new Hash256(reader.readUint256()), new UInt64Reader().read(reader));
+        return new Authority(reader.readUint256(), new UInt64Reader().read(reader));
```

Comment:
We use differente readers other that UInt64Reader as well. We are able to do that, but we might want to add others as well. Currently to avoid a lot of initializations of readers, we decided to make most of our readers Signleton with in constructor initialized param for reused readers, but this one is not converted yet.

Reply:
Modifying external libraries comes with the cost of then maintaining forks. I find the public API of `io.emeraldplay.polkaj.scale` fairly apt, so I'd say it's our responsibility to define and properly structure whatever en/decoding logic we need. And that's a valid point, scale en/decoding it's currently not very unified across the entire project.